### PR TITLE
Add shared code artifacts for UDF skills evals [skip ci]

### PR DESCRIPTION
**Pull request by rishic3**
Contributes to #15017, a step toward #15015 (publishing to NVIDIA/skills).

### Description

Adds `skills/evals/resources/`, containing code fixtures provided to the UDF skill evaluations.

A follow-up PR with the full evals will wire these fixtures into each skill's `evals/` directory.

#### Why

(#15017 provides some context, repeated briefly here).
The NVIDIA/skills CI requires each skill to have its own `evals/`, which defines a CI job that runs an agent with the skills and grades the outputs.

Several of the skills require artifacts from a previous step - for instance, `udf-convert-to-{x}` expects a completed unit test, `udf-benchmark` expects a working GPU implementation to benchmark. And of course, all skills require a CPU UDF to work with.

This PR creates adds these pre-provided artifacts so that each skill has something to work from. The artifacts will be placed into the agent's workspace during skill evals (wired in a follow-up PR).

#### What

Initially, this adds resources for a single UDF, `array-segment-sum`:
- `common/ArraySegmentSumUDF.scala` - a representative CPU Spark UDF.
- `common/UnitTest.scala` - a CPU unit test covering representative test cases and edge cases.
- `common/ArraySegmentSumRapidsUDF.scala` - a cuDF RapidsUDF, implementation, functionally equivalent to the CPU UDF.
- `judge/UnitTest.scala` - a *deliberately weak unit test* that is missing key edge cases (e.g. negative values, empty arrays, zeroes, etc.), specifically designed for the judge skill to flag.
- `judge/CudfComparisonTest.scala` - a *deliberately incorrect comparison test* that runs in `explainOnly` mode so the GPU path never actually executes, specifically designed for the judge skill to flag.

The UDF is intended to be non-trivial enough that the evaluation has meaningful signal. In the future we can expand the artifacts to cover more diverse input classes. 

### Checklists

Documentation
- [X] Updated for new or modified user-facing features or behaviors
- [ ] No user-facing change

Testing
- [ ] Added or modified tests to cover new code paths
- [ ] Covered by existing tests
      (Please provide the names of the existing tests in the PR description.)
- [X] Not required

Performance
- [ ] Tests ran and results are added in the PR description
- [ ] Issue filed with a link in the PR description
- [X] Not required

### CODE REVIEW

File: skills/evals/resources/array-segment-sum/common/UnitTest.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  override def beforeAll(): Unit = {
+    spark = SparkSession.builder()
+      .appName("UDF Unit Test")
+      .master("local[4]")
+      .config("spark.plugins", "com.nvidia.spark.SQLPlugin")
+      .config("spark.rapids.skipGpuArchitectureCheck", "true")
+      .config("spark.rapids.sql.mode", "explainOnly")
+      .config("spark.sql.adaptive.enabled", "false")
+      .enableHiveSupport()
```
[the remark this answers is not in the document]
Comment (rishic3) [reply skills/evals/resources/array-segment-sum/common/UnitTest.scala:97]: That's fine, the unit test is just for the CPU UDF. But I've filed https://github.com/NVIDIA/cudf-spark/issues/15173 to drop explainOnly mode (and for the unit test, the plugin entirely) to make this more explicit.

File: skills/evals/resources/array-segment-sum/judge/CudfComparisonTest.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    // Run CPU UDF
+    UnitTest.registerUDF(spark, "array_segment_sum_cpu")
+    val cpuResultDF = UnitTest.executeUDF(spark, "array_segment_sum_cpu", testDF)
+    UnitTest.verifyUDFResults(cpuResultDF, testDF)
+
+    // Run RapidsUDF
+    registerRapidsUDF(spark, "array_segment_sum_gpu")
+    val gpuResultDF = UnitTest.executeUDF(spark, "array_segment_sum_gpu", testDF)
+    UnitTest.verifyUDFResults(gpuResultDF, testDF)
+    TestUtils.assertDataFrameEquals(actual = gpuResultDF, expected = cpuResultDF)
```
[the remark this answers is not in the document]
Comment (rishic3) [reply skills/evals/resources/array-segment-sum/judge/CudfComparisonTest.scala:47]: Actually, the judge is not expected to run anything. It is just a static reviewer. It is true that there is some missing context, but once the evals are in the judge will have `gen-test` available as a skill so technically it can pull the full project if it wants to. I think it is minor enough not to be worth duplicating TestUtils again in resources/.

**Comment by nvauto** [comment]
**NOTE: release/26.08 has been created from main. Please retarget your PR to release/26.08 if it should be included in the release.**

**Comment by rishic3** [comment]
Hi @wjxiz1992 @sayedbilalbari, quick ping - would you mind reviewing or giving feedback on this? It breaks up the diff for https://github.com/NVIDIA/cudf-spark/pull/15184. 

We could also just close this and go straight for #15184, if preferable.
